scripts/criu-ns

```diff
@@ -102,6 +102,7 @@ def wrap_restore():
 
     criu_pid = os.fork()
     if criu_pid == 0:
+        os.setsid()
```

Comment:
Hi @Snorch, this change is results in the following error:

```console
cat > test.sh <<-EOF
#!/bin/sh
while : ; do date; sleep 1; done
EOF
chmod +x test.sh
./test.sh
```

```console
mkdir /tmp/test
sudo criu dump -t `pgrep test.sh` -D /tmp/test -j -v4
sudo criu-ns restore -D /tmp/test -j -v4
```

Restore error:
```
(00.003168)  16390: tty: open driver pts id 0x7 index 5 (master 0 sid 16105 pgrp 16390 inherit 1)
(00.003172)  16564: 	Receive fd for 2
(00.003191)  16390: tty: Migrated slave peer 0x7 -> to fd 0
(00.003222)  16390: Error (criu/tty.c:689): tty: Failed to set group 16390 on 0: Inappropriate ioctl for device
(00.003229)  16390: Error (criu/files.c:1200): Unable to open fd=0 id=0x7
(00.003272) Error (criu/cr-restore.c:2470): Restoring FAILED.
```

If I comment out this line `criu-ns restore` works as expected.

Reply:
Do you know why CI did not catch this?

Reply:
> Do you know why CI did not catch this?

Good question, I think we don't currently have CI tests for `criu-ns`.

Reply:
That would explain it 😁

Reply:
Thanks for catching this, I will try looking closer on it, but not sure if there is good solution.

Difference of your example (from my test with setsid() added) is that your test.sh creates root_item with sid inherited but  with own pgid. Without setsid() on restore pgid would be 0 that was probably a reason why problem was hidden...

If I do:
```
@@ -1043,8 +1045,7 @@ static int pty_open_unpaired_slave(struct file_desc *d, struct tty_info *slave)
                 */
                if (root_item->sid != vpid(root_item)) {
                        if (root_item->pgid == vpid(root_item)) {
-                               if (tty_set_prgp(fd, root_item->pgid))
-                                       goto err;
+                               ;
                        } else {
                                pr_debug("Restore inherited group %d\n", getpgid(getppid()));
                                if (tty_set_prgp(fd, getpgid(getppid())))
```
It helps at least on first dump/restore.

But second dump gives:
```
(00.001391) Error (criu/mount.c:927): mnt: Mount 1623 ./var/lib/nfs/rpc_pipefs (master_id: 566 shared_id: 0) has unreachable sharing. Try --enable-external-masters.
```
Which is probably another unrelated problem of criu-ns script, external mounts get to our criu-ns container when we do unshare(CLONE_NEWNS) =)

I think that adding setsid() is not a bug, but it reveals that we have some problem here in some cases...